app/src/main/java/com/bunbeauty/papakarlo/feature/auth/screen/login/LoginViewModel.kt

```diff
@@ -115,25 +128,41 @@ class LoginViewModel(
     }
 
     fun onNextClick() {
-        mutableIsLoading.value = true
+        mutableLoginState.update { oldState ->
+            oldState.copy(
+                state = LoginState.State.Loading
+            )
+        }
     }
 
     fun onSuccessVerified() {
-        viewModelScope.launch {
+        viewModelScope.launch(exceptionHandler) {
             userInteractor.login(
                 firebaseUserUuid = firebaseAuthRepository.firebaseUserUuid,
                 firebaseUserPhone = firebaseAuthRepository.firebaseUserPhone
             )
 
             when (successLoginDirection) {
-                BACK_TO_PROFILE -> router.navigate(LoginFragmentDirections.backToProfileFragment())
-                TO_CREATE_ORDER -> router.navigate(LoginFragmentDirections.toCreateOrderFragment())
+                BACK_TO_PROFILE -> mutableLoginState.update { state ->
+                    state.copy(
+                        eventList = state.eventList + LoginState.Event.NavigateBackToProfileFragment
+                    )
+                }
+                TO_CREATE_ORDER -> mutableLoginState.update { state ->
+                    state.copy(
+                        eventList = state.eventList + LoginState.Event.NavigateToCreateOrderFragment
+                    )
+                }
             }
         }
     }
 
     fun onVerificationError(error: String) {
-        mutableIsLoading.value = false
+        mutableLoginState.update { oldState ->
+            oldState.copy(
+                state = LoginState.State.Success
```

Comment:
Setting success state in case of verification error... Maybe show error

Reply:
need conversation